packages/playground/src/index.ts

```diff
@@ -0,0 +1,283 @@
+import { getGentraceApiKey } from "@gentrace/core";
+import { WebSocket } from "ws";
+import { v4 as uuidv4 } from "uuid";
+
+export { init } from "@gentrace/core"; // for accessing the Gentrace API key
+
+import { AsyncLocalStorage } from "async_hooks";
+const asyncLocalStorage = new AsyncLocalStorage();
+
+interface customObject {
+  typeName: string;
+  objectName: string;
+  object: object;
+}
+
+interface interactionObject {
+  name: string;
+  inputFields: object;
+  outputFields: object;
+  interaction: any;
+}
+
+interface stepInputObject {
+  provider: {
+    type: string;
+    default: string;
+    value: string;
+  };
+  prompt: {
+    type: string;
+    default: string;
+    value: string;
+  };
+}
+
+export class GentraceSession {
+  registeredCustomTypes: string[] = [];
+  registeredCustomObjects: customObject[] = [];
+  registeredInteractionObjects: interactionObject[] = [];
+
+  //WEBSOCKET_URL = "ws://localhost:8080";
+  WEBSOCKET_URL = "ws://localhost:3001";
+
+  // format customObjects for sending to WebSocket server
+  private formatSetupTypes(customObjects: customObject[]): object {
+    // create a Map to group customObjects by its typeName
+    const groupedByMap = customObjects.reduce((acc, item) => {
+      // check if the typeName already exists in the accumulator
+      if (!acc.has(item.typeName)) {
+        acc.set(item.typeName, []); // initialize the group as needed
+      }
+
+      // add the objectName to its typeName group
+      acc.get(item.typeName)?.push(item.objectName);
+
+      return acc;
+    }, new Map<string, string[]>());
+
+    // convert the Map to a formatted array and return it
+    let formattedArray: Object[] = [];
+
+    for (const [typeName, objectNames] of groupedByMap) {
+      formattedArray.push({
+        name: typeName,
+        objects: objectNames,
+      });
+    }
+
+    return formattedArray;
+  }
+
+  private isJson(message: string): boolean {
+    try {
+      JSON.parse(message);
+    } catch (e) {
+      return false;
+    }
+    return true;
+  }
+
+  // send registered objects to the WebSocket server to receive a playground URL
+  // and listen for messages from the server to run an interaction
+
+  public start(): void {
+    console.log("GENTRACE_API_KEY: " + getGentraceApiKey());
+
+    const setupEvent = {
+      id: uuidv4(),
+      init: "pg-sdk",
+      data: {
+        type: "setup",
+        apiKey: getGentraceApiKey(),
+        interactions: this.registeredInteractionObjects,
+        types: this.formatSetupTypes(this.registeredCustomObjects),
+      },
+    };
+
+    const ws = new WebSocket(this.WEBSOCKET_URL);
```

Comment:
I don't know if @dsafreno implemented a ping event on the server-side, but I've run into the issue documented by the below link a few times before when working with WebSockets. This isn't a blocker but probably appropriate for GA.

https://github.com/websockets/ws#how-to-detect-and-close-broken-connections

Reply:
[this commit](https://github.com/gentrace/gentrace-node/pull/341/commits/265178c9ba8b1b2fa8296f3e23f1a90a72b6394e) has a ping version that @dsafreno had used. I tested it and it seems to keep the connection from breaking too quickly on Staging.